src/main/java/gut/follower/org/Controller/LoginController.java

```diff
@@ -0,0 +1,35 @@
+package gut.follower.org.Controller;
+
+import gut.follower.org.Models.Account;
+import gut.follower.org.Repositories.AccountRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.security.Principal;
+import java.util.Map;
+
+/**
+ *
+ */
+@RestController
+@RequestMapping("/login")
+public class LoginController {
+
+    @Autowired
+    AccountRepository accountRepository;
+
+    @RequestMapping(method = RequestMethod.POST)
+    public Account authenticateUser(@RequestBody Map<String, String> map){
+        Account account = accountRepository.findByUsername(map.get("username"));
+        if(account != null && account.getPassword().equals(map.get("password"))){
```

Comment:
You can section this "if condition" off to make it more readable. For instance create
private boolean validatePassword(Account account) method. It would be neat to use Java 8 Optional class to avoid comparison like `if sth == null'. More on this topic [**here**](http://www.oracle.com/technetwork/articles/java/java8-optional-2175753.html).